plugins/pytorch/sam2_refiner.py

```diff
@@ -222,6 +222,19 @@ def refine(self, image_data, detections) -> DetectedObjectSet:
             # Extract the new mask info relative to the vital box
             box = vital_box_to_kwiamge(vital_det.bounding_box)
             sl = box.quantize().to_slice()
+
+            # Shift binmask and slice if negative slice start value
+            if sl[0].start < 0:
+                shift = -sl[0].start
+                binmask = np.roll(binmask, shift, axis=0)
+                binmask[:shift] = np.zeros((binmask[:shift].shape))
+                sl = (slice(0, sl[0].stop - sl[0].start), sl[1])
+            if sl[1].start < 0:
+                shift = -sl[1].start
+                binmask = np.roll(binmask, shift, axis=1)
+                binmask[:,:shift] = np.zeros((binmask[:,:shift].shape))
+                sl = (sl[0], slice(0, sl[1].stop - sl[1].start))
```

Comment:
```suggestion
            # use delayed image's padded negative slice functionality. 
            import delayed_image
            delayed = delayed_image.DelayedIdentity(binmask)
            binmask = delayed.crop(sl, clip=False, wrap=False).finalize()
```

Let's use this code instead. If for some reason we don't want to use delayed image, we should abstract this logic out as a helper function and fix the bottom right corner case. But it probably just makes more sense to use delayed image here.

Reply:
Note: the suggestion isn't quite right, it doesn't replace the next line where `relative_submask` is assigned. Hopefully it is clear what I meant. Can you fix this up and test it?

Reply:
Just applied your suggestion ! 
I also had to reshape `relative_submask` otherwise it crashed later when converting to PIL array. (A third dimension of size 1 was appended).

Works great now !